app/components/channel_system_message/channel_system_message.js

```diff
@@ -0,0 +1,239 @@
+                    style={style.baseText}
+                    navigator={navigator}
+                    textStyles={textStyles}
+                    theme={theme}
+                    values={{
+                        username,
+                        old: oldDisplayName,
+                        new: newDisplayName,
+                    }}
+                />
+            );
+        };
+
+        function renderPurposeChangeMessage(post) {
+            if (!post.props.username) {
+                return null;
+            }
+            const username = renderUsername(post.props.username);
+            const oldPurpose = post.props.old_purpose;
+            const newPurpose = post.props.new_purpose;
+            if (post.props.new_purpose) {
+                if (post.props.old_purpose) {
+                    return (
+                        <FormattedMarkdownText
+                            id='app.channel.post_update_channel_purpose_message.updated_from'
+                            defaultMessage='{username} updated the channel purpose from: {old} to: {new}'
+                            style={style.baseText}
+                            navigator={navigator}
+                            textStyles={textStyles}
+                            theme={theme}
+                            values={{
+                                username,
+                                old: oldPurpose,
+                                new: newPurpose,
+                            }}
+                        />
+                    );
+                }
+                return (
+                    <FormattedMarkdownText
+                        id='app.channel.post_update_channel_purpose_message.updated_to'
+                        defaultMessage='{username} updated the channel purpose to: {new}'
+                        style={style.baseText}
+                        navigator={navigator}
+                        theme={theme}
+                        textStyles={textStyles}
+                        values={{
+                            username,
+                            new: newPurpose,
+                        }}
+                    />
+                );
+            } else if (post.props.old_purpose) {
+                return (
+                    <FormattedMarkdownText
+                        id='app.channel.post_update_channel_purpose_message.removed'
+                        defaultMessage='{username} removed the channel purpose (was: {old})'
+                        style={style.baseText}
+                        navigator={navigator}
+                        textStyles={textStyles}
+                        theme={theme}
+                        values={{
+                            username,
+                            old: oldPurpose,
+                        }}
+                    />
+                );
+            }
+            return null;
+        }
+
+        return systemMessageRenderers[post.type](post);
```

Comment:
We might want a null check here on `systemMessageRenderers[post.type]` just to be safe